hera_pspec/uvpspec.py

```diff
@@ -394,6 +444,63 @@ def get_spw_ranges(self, spws=None):
 
         return spw_ranges
 
+    def get_stats(self, stat, key, *args):
+        """
+        Returns a statistic from the stats_array dictionary.
+
+        Parameters
+        ----------
+        stat: string
+            The statistic to return.
+
+        spw: int
+            Choice of spectral window.
+        """
+        if not hasattr(self, "stats_array"):
+            raise AttributeError("No stats have been entered to this UVPSpec object")
+
+        assert stat in self.stats_array.keys(), "Statistic name not found in stat keys."
+
+        spw, blpairts, pol = self.key_to_indices(key, *args)
+        data = self.stats_array[stat]
+        return data[spw][blpairts, :, pol]
+
+    def set_stats(self, stat, key, statistic, *args):
+        """
+        Sets a statistic in the stats_array.
+
+        Parameters
+        ----------
+        stat: string
+            Name of the statistic.
+
+        spw: int
+            Spectral window of the statistic.
+
+        statistic: ndarray
+            Array with statistics to set. Must be same shape as the same slice
+            of data_array.
+        """
+        spw, blpairts, pol = self.key_to_indices(key, *args)
+        statistic = np.asarray(statistic)
+
+        if self.data_array[spw][blpairts, :, pol].shape != statistic.shape:
+            errmsg = ("Input array (shape %r) must match the shape of data_array"
+                      "spectra (shape (%i, %i)).") % (statistic.shape,
+                                                      self.Ntimes,
+                                                      self.Ndlys)
+            raise ValueError(errmsg)
+
+        if not hasattr(self, "stats_array"):
+            self.stats_array = odict()
+
+        dtype = statistic.dtype
+        if stat not in self.stats_array.keys():
+            self.stats_array[stat] = odict([[i, np.nan * np.ones(self.data_array[i].shape, dtype=dtype)]
+                                             for i in range(self.Nspws)])
+
+        self.stats_array[stat][spw][blpairts, :, pol] = statistic
```

Comment:
we should add a conditional to catch if the spectra have been folded. see the `get_data` function for examples